test/unit/org/apache/cassandra/index/sai/memory/VectorMemtableIndexTest.java

```diff
@@ -141,11 +142,15 @@ public void randomQueryTest() throws Exception
             Set<Integer> foundKeys = new HashSet<>();
             int limit = getRandom().nextIntBetween(1, 100);
 
-            try (RangeIterator iterator = memtableIndex.search(new QueryContext(), expression, keyRange, limit))
+            try (var iterator = memtableIndex.orderBy(new QueryContext(), expression, keyRange, limit))
             {
+                ScoredPrimaryKey lastKey = null;
                 while (iterator.hasNext())
                 {
-                    PrimaryKey primaryKey = iterator.next();
+                    ScoredPrimaryKey primaryKey = iterator.next();
+                    if (lastKey != null)
+                        assertTrue( "Returned keys are not ordered by score", lastKey.score >= primaryKey.score);
```

Comment:
@jbellis - this assertion fails when we get vectors from after the resume search feature on the vector index. As I think about the approximate part of the search, this result seems expected. The main question is whether we should we use that knowledge to do something similar to the brute force calculations so that before we exhaust the previous search result, we call resume search, and put the results into a PriorityQueue to smooth out discontinuities? The resume logic is in `AutoResumingNodeScoreIterator`.

Reply:
You can find a better score if you search deeper after 100 results, 1 result, or 0 results, there's no way to prevent this 100% without scanning the entire graph.

Reply:
Makes sense. I'll fix the test.